satpy/composites/__init__.py

```diff
@@ -1701,3 +1702,85 @@ def __call__(self, projectables, nonprojectables=None, **info):
 
         masked_projectable = projectable.where(lon_min_max)
         return super().__call__([masked_projectable], **info)
+
+
+class MissingTime(Exception):
+    """Raised when temporal composite building lacks time dimension."""
+
+
+class BaseTemporalCompositor(CompositeBase):
+    """Compositors combining multiple time steps.
+
+    Base class for compositors that combine inputs from two or more time steps.
+
+    To use this, the user must start with a :class:`MultiScene` and load the
+    temporal composite from there.  Composite generation will fail due to missing
+    temporal information in the containing scenes, but it will add the temporal
+    composite to the scene wishlists.  Now we run ``MultiScene.blend(timeseries)``,
+    which will create DataArrays with a time dimension.  After blending, the
+    blend method tries again to generate the composites, and on this second run
+    the generation should work.
+    """
+
+    def __init__(self, name, prerequisites, optional_prerequisites=None, **kwargs):
+        """Initialise a temporal compositor."""
+        self._ensure_prerequisites_have_times(prerequisites)
+        super().__init__(name, prerequisites, optional_prerequisites, **kwargs)
+
+    def _check_time_dimension(self, projectables):
+        """Make sure all projectables have a time dimension."""
+        for projectable in projectables:
+            if "time" not in projectable.dims:
+                raise MissingTime(
+                    "Creating temporal composite needs time dimensions. "
+                    "Typically, this comes from starting with a MultiScene "
+                    "and then performing timeseries blending.")
+
+    def _ensure_prerequisites_have_times(self, prerequisites):
+        """Make sure all prerequisites have times defined."""
+        for preq in prerequisites:
+            if "time" not in preq.to_dict():
+                raise KeyError(
+                    "In a temporal composite, all prerequisites should have "
+                    "time information defined.  Time information is missing for "
+                    f"{preq!s}")
```

Comment:
Should this also make sure that `pd.Timedelta` understands the time value? Or should that wait until `__call__` time as you have it now?